File: open-api-web/src/services/multi-api-system/userController.ts
// @ts-ignore
/* eslint-disable */
import { request } from '@umijs/max';

/** 此处后端没有提供注释 POST /user/add */
export async function addUser(body: API.UserDo, options?: { [key: string]: any }) {
  return request<API.ResponseEntityBoolean>('/user/add', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /user/current */
export async function getCurrentUser(options?: { [key: string]: any }) {
  return request<API.ResponseEntityUserVo>('/user/current', {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/delete */
export async function deleteUser(body: number, options?: { [key: string]: any }) {
  return request<API.ResponseEntityBoolean>('/user/delete', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/login */
export async function userLogin(body: API.UserLoginDto, options?: { [key: string]: any }) {
  return request<API.ResponseEntityUserVo>('/user/login', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/logout */
export async function userLogout(options?: { [key: string]: any }) {
  return request<API.ResponseEntityInteger>('/user/logout', {
    method: 'POST',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /user/mailCaptcha */
export async function getMailCaptcha(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getMailCaptchaParams,
  options?: { [key: string]: any },
) {
  return request<API.ResponseEntityBoolean>('/user/mailCaptcha', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/register */
export async function userRegister(body: API.UserRegisterDto, options?: { [key: string]: any }) {
  return request<API.ResponseEntityLong>('/user/register', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /user/search */
export async function searchUsers(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.searchUsersParams,
  options?: { [key: string]: any },
) {
  return request<API.ResponseEntityIPageUserVo>('/user/search', {
    method: 'GET',
    params: {
      ...params,
      userDto: undefined,
      ...params['userDto'],
      pageDto: undefined,
      ...params['pageDto'],
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/update */
export async function updateUser(body: API.UserDo, options?: { [key: string]: any }) {
  return request<API.ResponseEntityBoolean>('/user/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/update/self */
export async function updateUserSelf(body: API.UserDo, options?: { [key: string]: any }) {
  return request<API.ResponseEntityBoolean>('/user/update/self', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: open-api-system/src/main/java/com/example/mps/controller/UserController.java
package com.example.mps.controller;

import com.baomidou.mybatisplus.core.metadata.IPage;
import com.example.mps.common.ResponseEntity;
import com.example.mps.common.ResponseCode;
import com.example.mps.exception.BusinessException;
import com.example.multicommon.pojo.domain.UserDo;
import com.example.mps.pojo.dto.PageDto;
import com.example.mps.pojo.dto.user.UserDto;
import com.example.mps.pojo.vo.UserVo;
import com.example.mps.pojo.dto.user.UserLoginDto;
import com.example.mps.pojo.dto.user.UserRegisterDto;
import com.example.mps.service.UserService;
import com.example.mps.utils.CommonUtils;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import lombok.extern.slf4j.Slf4j;
import org.springframework.util.StringUtils;
import org.springframework.web.bind.annotation.*;

/**
 * @author pengYuJun
 */
@Slf4j
@RestController
@RequestMapping("/user")
public class UserController {

    @Resource
    private UserService userService;

    @PostMapping("/register")
    public ResponseEntity<Long> userRegister(@RequestBody UserRegisterDto userRegisterDto, HttpServletRequest request) {
        if (userRegisterDto == null) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "参数为空");
        }
        String loginName = userRegisterDto.getLoginName();
        String loginPwd = userRegisterDto.getLoginPwd();
        String checkPwd = userRegisterDto.getCheckPwd();
        String email = userRegisterDto.getEmail();
        String captcha = userRegisterDto.getCaptcha();
        if (CommonUtils.isAnyBlank(loginName, loginPwd, checkPwd, email, captcha)) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "参数为空");
        }
        long result = userService.userRegister(loginName, loginPwd, checkPwd, email,  captcha, request);
        return ResponseEntity.success(result);
    }

    @PostMapping("/login")
    public ResponseEntity<UserVo> userLogin(@RequestBody UserLoginDto userLoginDto, HttpServletRequest request) {
        if (userLoginDto == null) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "参数为空");
        }
        String loginName = userLoginDto.getLoginName();
        String loginPwd = userLoginDto.getLoginPwd();
        if (CommonUtils.isAnyBlank(loginName, loginPwd)) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "参数为空");
        }
        UserVo user = userService.userLogin(loginName, loginPwd, request);
        return ResponseEntity.success(user);
    }

    @PostMapping("/logout")
    public ResponseEntity<Integer> userLogout(HttpServletRequest request) {
        if (request == null) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "会话无效");
        }
        int result = userService.userLogout(request);
        return ResponseEntity.success(result);
    }

    @GetMapping("/current")
    public ResponseEntity<UserVo> getCurrentUser(HttpServletRequest request) {
        UserVo currentUser = userService.getCurrentUser(request);
        return ResponseEntity.success(currentUser);
    }

    @GetMapping("/search")
    public ResponseEntity<IPage<UserVo>> searchUsers(UserDto userDto, PageDto pageDto, HttpServletRequest request) {
        if (!userService.isAdmin(request)) {
            throw new BusinessException(ResponseCode.FORBIDDEN, "无权限访问");
        }
        IPage<UserVo> userPage = userService.searchUser(userDto, pageDto);
        return ResponseEntity.success(userPage);
    }

    @PostMapping("/delete")
    public ResponseEntity<Boolean> deleteUser(@RequestBody long id, HttpServletRequest request) {
        if (!userService.isAdmin(request)) {
            throw new BusinessException(ResponseCode.FORBIDDEN, "无权限访问");
        }
        if (id <= 0) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "删除请求id值无效");
        }
        boolean result = userService.removeById(id);
        if (result) {
            return ResponseEntity.success(true);
        }else {
            return ResponseEntity.failed("删除失败");
        }
    }

    @PostMapping("/update")
    public ResponseEntity<Boolean> updateUser(@RequestBody UserDo userDo, HttpServletRequest request) {
        if (!userService.isAdmin(request)) {
            throw new BusinessException(ResponseCode.FORBIDDEN, "无权限访问");
        }
        if (userDo.getId() == null || userDo.getId() <= 0) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "更新请求id值无效");
        }
        boolean result = userService.updateById(userDo);
        if (result) {
            return ResponseEntity.success(true);
        }else {
            return ResponseEntity.failed("更新失败");
        }
    }

    @PostMapping("update/self")
    public  ResponseEntity<Boolean> updateUserSelf(@RequestBody UserDo userDo, HttpServletRequest request) {
        //
        if (userDo == null || userDo.getId() == null || CommonUtils.areAllOtherFieldsNull(userDo, "id")) {
            throw new BusinessException(ResponseCode.PARAMS_ERROR, "更新请求参数无效");
        }
        UserVo currentUser = userService.getCurrentUser(request);
        if (!currentUser.getId().equals(userDo.getId())) {
            throw new BusinessException(ResponseCode.FORBIDDEN);
        }
        boolean result = userService.updateById(userDo);
        if (result) {
            return ResponseEntity.success(true);
        }else {
            return ResponseEntity.failed("更新失败");
        }
    }

    @PostMapping("/add")
    public ResponseEntity<Boolean> addUser(@RequestBody UserDo userDo, HttpServletRequest request) {
        if (!userService.isAdmin(request)) {
            throw new BusinessException(ResponseCode.FORBIDDEN, "无权限访问");
        }
        boolean result = userService.addUser(userDo);
        if (result) {
            return ResponseEntity.success(true);
        }else {
            return ResponseEntity.failed("新增失败");
        }
    }

    @GetMapping("mailCaptcha")
    public ResponseEntity<Boolean> getMailCaptcha(String email, HttpServletRequest request) {
        String captcha = userService.getMailCaptcha(email, request);
        if (StringUtils.hasText(captcha)){
            request.getSession().setAttribute("captcha", captcha);
            return ResponseEntity.success(true);
        }else {
            return ResponseEntity.failed("获取验证码失败");
        }
    }
}
